app-typescript/components/Layouts/Panel.tsx

```diff
@@ -47,11 +47,16 @@ interface IPropsPanelHeader {
     theme?: 'light-ui' | 'dark-ui'; // defaults to 'light
     className?: string;
     onClose?(): void;
+    onPin?(): void;
+    pinned?: boolean;
 }
 
-class PanelHeader extends React.PureComponent<IPropsPanelHeader> {
+class PanelHeader extends React.PureComponent<IPropsPanelHeader, {rotate: boolean}> {
```

Comment:
Make an `IState` interface instead of inlining the rotate prop so we have consistency.